eif_utils/src/lib.rs

```diff
@@ -521,15 +671,58 @@ impl<T: Digest + Debug + Write + Clone> EifBuilder<T> {
         if self.sign_info.is_some() {
             self.generate_eif_signature(&measurements);
         }
+        self.save_metadata().expect("Failed to save metadata");
         self.compute_crc();
         self.write_header(output_file);
         self.write_kernel(output_file);
         self.write_cmdline(output_file);
         self.write_ramdisks(output_file);
         self.write_signature(output_file);
+        self.write_metadata(output_file);
         measurements
     }
 
+    pub fn get_generated_meta(&mut self) -> BTreeMap<String, String> {
+        let mut meta = BTreeMap::new();
+
+        let time = SystemTime::now();
+        let time: DateTime<Utc> = time.into();
+        meta.insert("BuildTime".to_string(), time.to_rfc3339());
+
+        // Get package version and name from CARGO_PKG_VERSION and CARGO_PKG_NAME
+        let tool_name = env!("CARGO_PKG_NAME").to_string();
```

Comment:
It puts the name and version of the current package: `eif_utils` while probably expected to put the root `nitro_cli` package details

Reply:
That's also the reason for unit tests failure

Reply:
Moved to a separate `EifBuildInfo` struct which is propagated as argument. `generate_build_info!` macros can be used to generate it based on current cargo package